File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/UserDeletionCheck.java
package de.sovity.authorityportal.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Result of the user deletion check.")
public class UserDeletionCheck {
    @Schema(description = "User's ID", requiredMode = Schema.RequiredMode.REQUIRED)
    String userId;
    @Schema(description = "Indicator if the user can be deleted", requiredMode = Schema.RequiredMode.REQUIRED)
    Boolean canBeDeleted;
    @Schema(description = "Indicator for the user being the last PA in their organization", requiredMode = Schema.RequiredMode.REQUIRED)
    Boolean isLastParticipantAdmin;
    @Schema(description = "Indicator for the user being the creator of their organization", requiredMode = Schema.RequiredMode.REQUIRED)
    Boolean isOrganizationCreator;
    @Schema(description = "List of possible successors (if needed)", requiredMode = Schema.RequiredMode.REQUIRED)
    List<PossibleCreatorSuccessor> possibleSuccessors;
}

File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/UserDeletionCheck.kt
/*
 * Copyright (c) 2024 sovity GmbH
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contributors:
 *      sovity GmbH - initial implementation
 */

package de.sovity.authorityportal.api.model

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Result of the user deletion check.")
data class UserDeletionCheck(
    @field:Schema(description = "User's ID", requiredMode = Schema.RequiredMode.REQUIRED)
    val userId: String,

    @field:Schema(description = "Indicator if the user can be deleted", requiredMode = Schema.RequiredMode.REQUIRED)
    val canBeDeleted: Boolean,

    @field:Schema(
        description = "Indicator for the user being the last PA in their organization",
        requiredMode = Schema.RequiredMode.REQUIRED,
    )
    @JsonProperty("isLastParticipantAdmin") // Workaround because codegen messes up boolean naming
    val isLastParticipantAdmin: Boolean,

    @field:Schema(
        description = "Indicator for the user being the creator of their organization",
        requiredMode = Schema.RequiredMode.REQUIRED,
        name = "isOrganizationCreator"
    )
    @JsonProperty("isOrganizationCreator") // Workaround because codegen messes up boolean naming
    val isOrganizationCreator: Boolean,

    @field:Schema(description = "List of possible successors (if needed)", requiredMode = Schema.RequiredMode.REQUIRED)
    var possibleCreatorSuccessors: MutableList<PossibleCreatorSuccessor>,
)
